Backend Code/VigyBag/users/serializers.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework import serializers
+from .models import CustomUser
+from django.contrib.auth import authenticate
+
+class RegisterUserSerializer(serializers.ModelSerializer):
+    email=serializers.EmailField(max_length=255)
+    password=serializers.CharField(max_length=128,write_only=True)
+
+    class Meta:
+        model = CustomUser
+        fields = ('email', 'password')  
+        #have to add usertype ,default customer
+    
+    def create(self, validated_data):
+        user = CustomUser.objects.create_user(email=validated_data['email'], password=validated_data['password'])
+        return user
```

Comment:
This is a Create operation (An operation that should ideally be performed by repository of the code). It should not be placed in the serializer file. 

Serializers allow complex data such as querysets and model instances to be converted to native Python datatypes that can then be easily rendered into JSON, XML or other content types. Serializers also provide deserialization, allowing parsed data to be converted back into complex types, after first validating the incoming data.

Ref: https://www.django-rest-framework.org/api-guide/serializers/